### ISSUE TITLE
[ads] Follow up to #38191: Restoring window does not resume landed counter

### ISSUE THREAD

**Issue by btlechowski** [brave/brave-browser#38234]
Follow up to: https://github.com/brave/brave-browser/issues/38191

## Steps to Reproduce

1. Clean profile
2. Trigger ad
3. Start landed counter
4. Minimize the browser window
5. Restore the browser window

Note: the counter was suspended on minimize.

## Actual result:

Landed counter is not resumed

## Expected result:
Landed counter is resumed

## Reproduces how often: 

Easily reproduced

## Brave version (brave://version info)

Brave | 1.68.6 Chromium: 125.0.6422.41 (Official Build) nightly (64-bit)
-- | --
Revision | 2886cb2a986d182705496c6252456cf7424e08a5
OS | Linux

cc @tmancey @aseren @iambrianfung

**Comment by kjozwiak** [comment]
The above requires `1.67.100` or higher for `1.67.x` verification 👍

**Comment by btlechowski** [comment]
Verified with

Brave | 1.67.104 Chromium: 126.0.6478.26 (Official Build) beta (64-bit)
-- | --
Revision | 9a4d1cb2f47a60953416cd3f39f0520e4dd4cf0e
OS | Linux


#### minimized & restored

<details>

```
[4902:4902:0604/155524.632319:VERBOSE1:browser_manager.cc(53)] Browser did become active
[4902:4902:0604/155524.660808:VERBOSE2:tab_manager.cc(259)] Tab id 445217042 did become focused
[4902:4902:0604/155524.665779:VERBOSE1:browser_manager.cc(76)] Browser did enter foreground
[4902:4902:0604/155524.668184:VERBOSE1:ad_handler.cc(156)] Resumed page landing on tab id 445217042 and maybe land in 4.92957 s
[4902:4902:0604/155525.915484:VERBOSE1:browser_manager.cc(53)] Browser did resign active
[4902:4902:0604/155525.917085:VERBOSE1:ad_handler.cc(150)] Suspended page landing on tab id 445217042 with 3.68039 s remaining
[4902:4902:0604/155525.918120:VERBOSE2:tab_manager.cc(259)] Tab id 445217042 did become occluded
[4902:4902:0604/155525.919524:VERBOSE1:browser_manager.cc(76)] Browser did enter background
```


</details>


#### background and foreground

<details>

```
[4902:4902:0604/155650.164018:VERBOSE1:browser_manager.cc(53)] Browser did become active
[4902:4902:0604/155650.168875:VERBOSE2:tab_manager.cc(259)] Tab id 445217042 did become focused
[4902:4902:0604/155650.171344:VERBOSE1:browser_manager.cc(76)] Browser did enter foreground
[4902:4902:0604/155650.174140:VERBOSE1:ad_handler.cc(156)] Resumed page landing on tab id 445217042 and maybe land in 0.994391 s
[4902:4902:0604/155650.892855:VERBOSE1:browser_manager.cc(53)] Browser did resign active
[4902:4902:0604/155650.896856:VERBOSE1:ad_handler.cc(150)] Suspended page landing on tab id 445217042 with 0.274508 s remaining
[4902:4902:0604/155650.897425:VERBOSE2:tab_manager.cc(259)] Tab id 445217042 did become occluded
[4902:4902:0604/155650.898594:VERBOSE1:browser_manager.cc(76)] Browser did enter background

```


</details>


#### switching tabs

<details>

```

[5583:5583:0604/155957.083111:VERBOSE2:tab_manager.cc(259)] Tab id 1010492463 did become occluded
[5583:5583:0604/155957.086946:VERBOSE2:tab_manager.cc(259)] Tab id 1010492464 did become focused
[5583:5583:0604/155957.089139:VERBOSE1:ad_handler.cc(156)] Resumed page landing on tab id 1010492464 and maybe land in 4.93243 s
[5583:5583:0604/160000.132795:VERBOSE2:tab_manager.cc(259)] Tab id 1010492464 did become occluded
[5583:5583:0604/160000.136343:VERBOSE1:ad_handler.cc(150)] Suspended page landing on tab id 1010492464 with 1.88529 s remaining
[5583:5583:0604/160000.136691:VERBOSE2:tab_manager.cc(259)] Tab id 1010492463 did become focused
[5583:5583:0604/160002.716781:VERBOSE2:tab_manager.cc(259)] Tab id 1010492463 did become occluded
[5583:5583:0604/160002.719718:VERBOSE2:tab_manager.cc(259)] Tab id 1010492464 did become focused
[5583:5583:0604/160002.720614:VERBOSE1:ad_handler.cc(156)] Resumed page landing on tab id 1010492464 and maybe land in 1.88529 s

```


</details>


#### landed event

<details>

```
[4902:4902:0604/155736.991820:VERBOSE1:redeem_non_reward_confirmation.cc(57)] Redeem non-reward confirmation
[4902:4902:0604/155736.992089:VERBOSE1:redeem_non_reward_confirmation.cc(66)] Create non-reward confirmation
[4902:4902:0604/155736.992219:VERBOSE6:redeem_non_reward_confirmation.cc(71)] URL Request:
  URL: https://anonymous.ads.bravesoftware.com/v3/confirmation/5192943b-b626-432e-8353-c6dca2cd9ad1
  Content: {"creativeInstanceId":"c1bbef73-5201-4ccd-9ed0-d897610d6a0e","transactionId":"5192943b-b626-432e-8353-c6dca2cd9ad1","type":"landed"}
  Content Type: application/json
  Method: kPost
[4902:4902:0604/155736.992567:VERBOSE7:redeem_non_reward_confirmation.cc(72)]   Headers:
    accept: application/json
[4902:4902:0604/155737.467470:VERBOSE6:redeem_non_reward_confirmation.cc(85)] URL Response:
  URL: https://anonymous.ads.bravesoftware.com/v3/confirmation/5192943b-b626-432e-8353-c6dca2cd9ad1
  Response Status Code: 418
  Response: {"id":"5192943b-b626-432e-8353-c6dca2cd9ad1","payload":{},"type":"landed","createdAt":"2024-06-04T14:12:03.612Z","modifiedAt":"2024-06-04T14:12:03.613Z","creativeInstanceId":"c1bbef73-5201-4ccd-9ed0-d897610d6a0e"}
[4902:4902:0604/155737.467837:VERBOSE7:redeem_non_reward_confirmation.cc(86)]   Headers:
    accept-ranges: bytes
    access-control-allow-credentials: true
    content-length: 213
    content-security-policy: sandbox
    content-type: application/json; charset=utf-8
    cross-origin-opener-policy: same-origin
    cross-origin-resource-policy: same-origin
    date: Tue, 04 Jun 2024 14:12:03 GMT
    referrer-policy: same-origin
    strict-transport-security: max-age=31536000; includeSubDomains; preload
    vary: Origin
    via: 1.1 varnish
    x-cache: MISS
    x-cache-hits: 0
    x-content-type-options: nosniff
    x-frame-options: DENY
    x-served-by: cache-hel1410034-HEL
[4902:4902:0604/155737.468176:VERBOSE1:redeem_non_reward_confirmation.cc(106)] Successfully redeemed non-reward landed confirmation for new_tab_page_ad with transaction id 5192943b-b626-432e-8353-c6dca2cd9ad1 and creative instance id c1bbef73-5201-4ccd-9ed0-d897610d6a0e
[4902:4902:0604/155737.479980:VERBOSE1:confirmations.cc(97)] Successfully processed landed confirmation for new_tab_page_ad with transaction id 5192943b-b626-432e-8353-c6dca2cd9ad1 and creative instance id c1bbef73-5201-4ccd-9ed0-d897610d6a0e

```

</details>
